### DISCUSSION TITLE
Hosting on cloud instance but can only access dashboard from local machine if I run `pocketbase serve --http="0.0.0.0:80"

### DISCUSSION THREAD

**Discussion by kunasun** [pocketbase/pocketbase#3839] [Q&A]
Having a strange issue when trying to set up pocketbase to run on an OCI instance. I would like to be able to access the dashboard from my local machine, but I am only able to do so if i run `./pocketbase serve --http="0.0.0.0:80`. 

If i add the https flag, or try to directly serve the ip of my cloud instance with `./pocketbase serve [ip of my instance]` I am unable to access the dashboard.

Additionally, If i try to run `./pocketbase serve --http="[ip of my instance]:80"` I get the following error: `listen tcp [instance ip]:80: bind: cannot assign requested address`. 

I have verified that nothing is listening on the port already, and the ports are forwarded as well as exposed in the firewall/oci ingress rules. I checked this by running a python3 http server on port 80 and I was able to access it just fine from my local machine. I also tried using the nginx config to point to the pocketbase server but I also had no luck with that. 

If anyone has run into this problem before, help would be appreciated. 

EDIT: I tried updating my /etc/hosts/ file to point both of the ips listed with `hostname -I` to my instance ip, but I am still unable to run `./pocketbase serve --http="[ip of my instance]:80"`

**Comment by ganigeorgiev** [comment]
Considering that it works with `0.0.0.0:80` (aka. your user is allowed to bind 80 port), then I suspect that probably your `/etc/hosts` file need to be updated.

Double check that you've extracted the correct `hostname -I` IP as usually it could return more than one entry (_I guess it could also depend on the type of network virtualization that OCI has but I'm not knowledgeable enough on this_).

But in general, if PocketBase is the only service that will run on your OCI instance, then it is safe to bind to `0.0.0.0`.

**Reply by kunasun** [reply]
`hostname -I` returns two ips, one is a `10.0.0.xx` and the other is a `172.xx.x.1` I added two lines to my `/etc/hosts` file, both of them in the format: `10.0.0.xx [my instance ip]`. 

Is that the correct way to update the /etc/hosts file? I am still unable to run `./pocketbase serve --http="[ip of my instance]:80"` without the bind error. I guess if all else fails, I will just bind to 0.0.0.0 but I might want to have other services running in the future. 

Interestingly enough, i can run `./pocketbase serve --http="10.0.0.xx:80"` it works and i can connect externally, but if I add the https flag I cannot connect externally even though there is no bind error. I still cannot use the ip of my instance even though the loopback address is in the /etc/hosts file. 

Another weird thing, if I have both the http and https flag, then when I try to curl the ip address of my instane, I get the following result:  `<a href="https://[ip of instance]/_">Found</a>.`

**Reply by ganigeorgiev** [reply]
I'm confused and I'm not sure that I'll be able to help you but about the question why the `--https` doesn't work with your IP - when `--https` flag is used PocketBase will try to generate a Let's encrypt certificate but the caveat is that Let's encrypt currently doesn't support issuing certificates for bare IPs, aka. you'll need a public valid domain name if you want to use Let's encrypt.

**Reply by ganigeorgiev** [reply]
Also about:

> _I will just bind to 0.0.0.0 but I might want to have other services running in the future._

It shouldn't matter whether you'll use `0.0.0.0` or the IP of the instance as in both cases it will fail if you try to bind multiple services to the same port. If you need to host more than 1 application on the 80/443 port my recommendation would be to put them behind a reverse proxy like nginx, apache, caddy, etc. (minimal example nginx config could be found at https://pocketbase.io/docs/going-to-production/#using-reverse-proxy).

**Reply by kunasun** [reply]
I've tried using the https flag with my personal domain that is pointing to my instance's ip and it also doesn't seem to work in that case, maybe I don't have the DNS setup correctly?

**Reply by ganigeorgiev** [reply]
I'm not really sure how to help you debug it.

Ideally if you have a domain pointing to your server, it should be enough to run only:

```sh
./pocketbase serve yourdomain.com
```

If you don't want to bind to `0.0.0.0` and to some explicit custom address, then you can further specify the `--http` and `--https` flags:

```sh
./pocketbase serve yourdomain.com --http="SOME_IP:80" --https="SOME_IP:443"
```
